### PR TITLE
docs: convert root storage daemon doc to markdown

### PR DESCRIPTION
**Pull request by poettering**
Import from:

https://www.freedesktop.org/wiki/Software/systemd/RootStorageDaemons/

### CODE REVIEW

**Comment by mrc0mmand** [comment]
The fail in `TEST-43-PRIVATEUSER-UNPRIV` is definitely not related to this PR, but I've seen it a few times in other PRs, not sure what's going on:

Console:
```
[[0;1;31mFAILED[0m] Failed to start [0;1;39mUser Manager for UID 4711[0m.
See 'systemctl status user@4711.service' for details.
```

journal:
```
Jan 13 19:28:43 systemd-testsuite systemd[1]: user@4711.service: Changed dead -> start
Jan 13 19:28:43 systemd-testsuite systemd[1]: Starting User Manager for UID 4711...
...
Jan 13 19:28:43 systemd-testsuite systemd[1]: user@4711.service: User lookup succeeded: uid=4711 gid=4711
...
Jan 13 19:28:43 systemd-testsuite audit[304]: USER_ACCT pid=304 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:accounting grantors=pam_unix acct="testuser" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Jan 13 19:28:43 systemd-testsuite systemd[304]: pam_warn(systemd-user:setcred): function=[pam_sm_setcred] flags=0x2 service=[systemd-user] terminal=[<unknown>] user=[testuser] ruser=[<unknown>] rhost=[<unknown>]
Jan 13 19:28:43 systemd-testsuite systemd[304]: PAM failed: Failure setting user credentials
Jan 13 19:28:43 systemd-testsuite audit[304]: CRED_ACQ pid=304 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=? acct="testuser" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
Jan 13 19:28:43 systemd-testsuite systemd[304]: user@4711.service: Failed to set up PAM session: Operation not permitted
Jan 13 19:28:43 systemd-testsuite systemd[304]: user@4711.service: Failed at step PAM spawning /usr/lib/systemd/systemd: Operation not permitted
Jan 13 19:28:43 systemd-testsuite systemd[1]: Received SIGCHLD from PID 304 ((systemd)).
Jan 13 19:28:43 systemd-testsuite systemd[1]: Child 304 ((systemd)) died (code=exited, status=224/PAM)
Jan 13 19:28:43 systemd-testsuite systemd[1]: user@4711.service: Child 304 belongs to user@4711.service.
Jan 13 19:28:43 systemd-testsuite systemd[1]: user@4711.service: Main process exited, code=exited, status=224/PAM
Jan 13 19:28:43 systemd-testsuite systemd[1]: user@4711.service: Failed with result 'protocol'.
Jan 13 19:28:43 systemd-testsuite systemd[1]: user@4711.service: Service will not restart (restart setting)
Jan 13 19:28:43 systemd-testsuite systemd[1]: user@4711.service: Changed start -> failed
Jan 13 19:28:43 systemd-testsuite systemd[1]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=88 reply_cookie=0 signature=sa{sv}a>
Jan 13 19:28:43 systemd-testsuite systemd[1]: user@4711.service: Job 76 user@4711.service/start finished, result=failed
Jan 13 19:28:43 systemd-testsuite systemd[1]: Failed to start User Manager for UID 4711.
...
Jan 13 19:28:43 systemd-testsuite systemd[1]: Got disconnect on private connection.
Jan 13 19:28:43 systemd-testsuite testsuite.sh[305]: + runas testuser systemd-run --user --unit=test-private-users -p PrivateUsers=yes -P echo hello
Jan 13 19:28:43 systemd-testsuite testsuite.sh[305]: + declare userid=testuser
Jan 13 19:28:43 systemd-testsuite testsuite.sh[305]: + shift
Jan 13 19:28:43 systemd-testsuite testsuite.sh[305]: + su testuser -s /bin/sh -c 'XDG_RUNTIME_DIR=/run/user/$UID exec "$@"' -- sh systemd-run --user --unit=test-private-users -p PrivateUsers=yes -P echo hello
Jan 13 19:28:43 systemd-testsuite systemd[1]: Got message type=method_call sender=:1.2 destination=org.freedesktop.systemd1 path=/org/freedesktop/systemd1/unit/user_2d4711_2eslice interface=org.freedesktop.DBus.Properties member=Get cooki>
Jan 13 19:28:43 systemd-testsuite systemd[1]: Sent message type=method_return sender=n/a destination=:1.2 path=n/a interface=n/a member=n/a cookie=194 reply_cookie=104 signature=v error-name=n/a error-message=n/a
Jan 13 19:28:43 systemd-testsuite systemd[1]: Got message type=method_call sender=:1.2 destination=org.freedesktop.systemd1 path=/org/freedesktop/systemd1/job/239 interface=org.freedesktop.DBus.Properties member=Get cookie=105 reply_cooki>
Jan 13 19:28:43 systemd-testsuite systemd[1]: Unprocessed message call sender=:1.2 object=/org/freedesktop/systemd1/job/239 interface=org.freedesktop.DBus.Properties member=Get
Jan 13 19:28:43 systemd-testsuite systemd[1]: Sent message type=error sender=n/a destination=:1.2 path=n/a interface=n/a member=n/a cookie=195 reply_cookie=105 signature=s error-name=org.freedesktop.DBus.Error.UnknownObject error-message=>
Jan 13 19:28:43 systemd-testsuite audit[308]: USER_AUTH pid=308 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:authentication grantors=pam_rootok acct="testuser" exe="/usr/bin/su" hostname=? addr=? terminal=? res=success'
Jan 13 19:28:43 systemd-testsuite audit[308]: USER_ACCT pid=308 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:accounting grantors=pam_unix acct="testuser" exe="/usr/bin/su" hostname=? addr=? terminal=? res=success'
Jan 13 19:28:43 systemd-testsuite su[308]: (to testuser) root on none
Jan 13 19:28:43 systemd-testsuite systemd[1]: Got message type=method_call sender=:1.4 destination=org.freedesktop.systemd1 path=/org/freedesktop/systemd1 interface=org.freedesktop.systemd1.Manager member=GetDynamicUsers cookie=2 reply_co>
Jan 13 19:28:43 systemd-testsuite systemd[1]: Sent message type=method_return sender=n/a destination=:1.4 path=n/a interface=n/a member=n/a cookie=196 reply_cookie=2 signature=a(us) error-name=n/a error-message=n/a
Jan 13 19:28:43 systemd-testsuite audit[308]: CRED_ACQ pid=308 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_rootok acct="testuser" exe="/usr/bin/su" hostname=? addr=? terminal=? res=success'
Jan 13 19:28:43 systemd-testsuite audit[308]: USER_START pid=308 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:session_open grantors=pam_unix acct="testuser" exe="/usr/bin/su" hostname=? addr=? terminal=? res=success'
Jan 13 19:28:43 systemd-testsuite su[308]: pam_unix(su:session): session opened for user testuser by (uid=0)
Jan 13 19:28:43 systemd-testsuite audit[308]: USER_END pid=308 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:session_close grantors=pam_unix acct="testuser" exe="/usr/bin/su" hostname=? addr=? terminal=? res=success'
Jan 13 19:28:43 systemd-testsuite audit[308]: CRED_DISP pid=308 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_rootok acct="testuser" exe="/usr/bin/su" hostname=? addr=? terminal=? res=success'
Jan 13 19:28:43 systemd-testsuite su[308]: pam_unix(su:session): session closed for user testuser
Jan 13 19:28:43 systemd-testsuite systemd[1]: systemd-journald.service: Received EPOLLHUP on stored fd 50 (stored), closing.
Jan 13 19:28:43 systemd-testsuite systemd[1]: testsuite.service: Control group is empty.
Jan 13 19:28:43 systemd-testsuite systemd[1]: Received SIGCHLD from PID 305 (testsuite.sh).
Jan 13 19:28:43 systemd-testsuite systemd[1]: Child 305 (testsuite.sh) died (code=exited, status=1/FAILURE)
Jan 13 19:28:43 systemd-testsuite systemd[1]: testsuite.service: Child 305 belongs to testsuite.service.
Jan 13 19:28:43 systemd-testsuite systemd[1]: testsuite.service: Main process exited, code=exited, status=1/FAILURE
Jan 13 19:28:43 systemd-testsuite systemd[1]: testsuite.service: Failed with result 'exit-code'.
Jan 13 19:28:43 systemd-testsuite systemd[1]: testsuite.service: Service will not restart (restart setting)
Jan 13 19:28:43 systemd-testsuite systemd[1]: testsuite.service: Changed start -> failed
Jan 13 19:28:43 systemd-testsuite systemd[1]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/systemd1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=197 reply_cookie=0 signature=sa{sv}>
Jan 13 19:28:43 systemd-testsuite systemd[1]: testsuite.service: Job 75 testsuite.service/start finished, result=failed
Jan 13 19:28:43 systemd-testsuite systemd[1]: Failed to start Testsuite service.
```

File: docs/ROOT_STORAGE_DAEMONS.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  that it is simply necessary to inform systemd about the socket you'd like it
+  to listen on on behalf of your daemon and minimally modify the daemon to
+  receive the listening socket for its services from systemd instead of
+  creating it on its own. Such modifications can be minimal, and are easily
+  written in a way that does not negatively impact usability on non-systemd
+  systems. For more information on making use of socket activation in your
+  program consult this blog story: [Socket
+  Activation](http://0pointer.de/blog/projects/socket-activation.html)
+
+* Consider having a look at the [initrd Interface of systemd](http://www.freedesktop.org/wiki/Software/systemd/InitrdInterface)
```
Comment (keszybz) [inline docs/ROOT_STORAGE_DAEMONS.md:193]: This will need to get updated...

**Comment by poettering** [comment]
@mrc0mmand hmm, that failure might be related to me merging #11199. We didn't use to call pam_setcred before.

Might have something to do with the stuff one can control with /proc/$PID/setgroups, but htat's just a wild guess

**Comment by poettering** [comment]
(#11199 did pass the CI though, but given that pam_setcred wasn't used before this has to do something with it)

**Comment by mrc0mmand** [comment]
Ah, I see, thanks. I just filed #14567 to keep track of this issue.

**Pull request merged**